### ISSUE TITLE
Feature request: add ability to easily extend the management of published dashboards

### ISSUE THREAD

**Issue by rs2** [jupyter-incubator/dashboards_server#288]
At the moment the home screen with a lain list of published dashboards can hardly be described as ready for the end user.

Consider:

1. Adding subfolders
2. Permissioning folders to users/user groups
3. Add the ability to rename and delete published dashboards
4. How to update the published dashboard when a new version of its dependency is released? (Think full SDLC, not just a snapshot in time).
5. Make the UI easily extendable/customizable through themes/CSS.

**Comment by parente** [comment]
@rs2 We'd be happy to have help with any and all of these. PRs welcome!

Some comments:

1. Subfolders are implemented IIRC, as long as the call to the API includes the subfolder path.
2. I think this is the idea behind #124 
3. Only the CLI handles this today. An admin UI that builds atop #124 would be good.
4. If a dependency changes, you need to repush the notebook bundle to update the copy on the server. This follows the immutable deploys practice.
5. Sure.

**Comment by dalogsdon** [comment]
> Subfolders are implemented IIRC, as long as the call to the API includes the subfolder path.

yes. It is a bit unclear since normally a folder will show with a folder icon but a folder containing a "bundled" dashboard looks just like a dashboard file in the listing UI. 

When uploading a dashboard file you can use a path like "some/path/to/dashboard.ipynb" and the intermediate folders "some", "path", "to" will be created.

**Comment by rs2** [comment]
> as long as the call to the API includes the subfolder path.

Where can I find the description of this API? Just using bundlers...

**Comment by dalogsdon** [comment]
https://github.com/jupyter-incubator/dashboards_server/wiki/Server-API#upload-notebook
